builds/checkin/watchdog.yaml

```diff
@@ -0,0 +1,54 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - iiot
+jobs:
+
+  ################################################################################
+  - job: check_run_pipeline
+  ################################################################################
+    displayName: Check pipeline preconditions (changes ARE in builds or edge hub)
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|edge-hub/watchdog)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes inside builds or edgehub folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+          fi
+        displayName: Check changes in sources
+        name: check_files
+
+  ################################################################################
+  - job: linux_amd64
+  ################################################################################
+    displayName: Linux amd64
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - script: echo "##vso[task.setvariable variable=RUST_BACKTRACE;]1"
+        displayName: Set env variables
+      - bash: scripts/linux/generic-rust/install.sh --project-root "edge-hub/watchdog"
+        displayName: Install Rust
+      - bash: scripts/linux/generic-rust/build.sh --project-root "edge-hub/watchdog" --packages "watchdog"
+        displayName: Build
+
+  ################################################################################
+  - job: style_check
```

Comment:
Btw, do we need to build this for arm? I think a separate style check job made sense when we build for multiple platforms (linux/windows, since there could be conditional code paths). Otherwise you can combine everything in one job.

Reply:
The checkin gates usually just build on amd64 since that is the arch of the agent. I think combining these is a good idea and will do so.

Reply:
Unfortunately, I cannot consolidate the jobs in the mqtt / edgelet / dotnet checkins because my other PR to remove all windows files is still under review. I will do so only for watchdog here then open a separate PR to clean up the others.